common/geofence/src/u_geofence.c

```diff
@@ -1258,21 +1260,18 @@ static uGeofencePositionState_t testSquareExtent(const uGeofenceSquare_t *pSquar
 static uGeofencePositionState_t testSpeed(const uGeofenceDynamic_t *pPreviousDistance)
 {
     uGeofencePositionState_t positionState = U_GEOFENCE_POSITION_STATE_NONE;
-    int32_t timeNowMs;
+    uint32_t timeDifferenceMs;
     int64_t distanceTravelledMillimetres;
 
     if ((pPreviousDistance->lastStatus.distanceMillimetres != LLONG_MIN) &&
         (pPreviousDistance->maxHorizontalSpeedMillimetresPerSecond >= 0)) {
         // Work out how far we can have travelled in the time
-        timeNowMs = uPortGetTickTimeMs();
-        // Guard against wrap
-        if (timeNowMs > pPreviousDistance->lastStatus.timeMs) {
-            // Divide by 1000 below to get per second
-            distanceTravelledMillimetres = ((int64_t) (timeNowMs - pPreviousDistance->lastStatus.timeMs)) *
-                                           pPreviousDistance->maxHorizontalSpeedMillimetresPerSecond / 1000;
-            if (distanceTravelledMillimetres < pPreviousDistance->lastStatus.distanceMillimetres) {
-                positionState = U_GEOFENCE_POSITION_STATE_OUTSIDE;
-            }
+        timeDifferenceMs = uTimeoutElapsedMs(pPreviousDistance->lastStatus.timeoutStart);
+        // Divide by 1000 below to get per second
+        distanceTravelledMillimetres = ((int64_t) timeDifferenceMs) *
```

Comment:
why int64_t and not uint64_t all values are positive

Reply:
Only because, when time was signed, it was simpler to keep everything signed. The structure member `lastStatus.distanceMillimetres`, which gets involved in this lower down, is signed and that structure is shared with the underlying Wi-Fi, cellular and GNSS code; since it is 64-bit anyway I think there should be enough room to keep it signed...?